marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping-functions/core.sjs

```diff
@@ -201,23 +201,34 @@ function isNullOrEmpty(value) {
 // END date/dateTime functions
 
 /**
- * Convenience function for getting the value of a key from a JSON object. An expression such as 
+ * Convenience function for getting the value of a key from a JSON object. An expression such as
  * "object['value']" doesn't work because 'value' is interpreted as an XPath predicate, not a value selector.
- * 
- * Note that "key" is the first argument for consistency with documentLookup and memoryLookup. 
- * 
- * @param key 
- * @param object 
- * @returns 
+ *
+ * Note that "key" is the first argument for consistency with documentLookup and memoryLookup.
+ *
+ * @param key
+ * @param object
+ * @returns
  */
 function lookup(key, object) {
   return object ? object[key] : null;
 }
 
+function hubURI(entityType) {
+  if(typeof entityType != "string"){
+    httpUtils.throwBadRequest(`Unable to generate uri: 'entityType' should be of type 'string'`);
```

Comment:
Can you verify that the Test capability in HC supports 400 errors, and also the generic Error that is thrown in other places in this library? That doesn't matter for this PR, just something to test.